pydantic/_internal/_generate_schema.py

```diff
@@ -1106,6 +1116,72 @@ def _get_or_cache_recursive_ref(self, cls: type[Any]) -> tuple[str, core_schema.
             self.recursion_cache[obj_ref] = core_schema.definition_reference_schema(obj_ref)
             return obj_ref, None
 
+    def _annotated_schema(self, annotated_type: Any) -> core_schema.CoreSchema:
+        """
+        Generate schema for an Annotated type, e.g. `Annotated[int, Field(...)]` or `Annotated[int, Gt(0)]`.
+        """
+        first_arg, *other_args = get_args(annotated_type)
+        return self.apply_annotations(CallbackGetCoreSchemaHandler(self.generate_schema), first_arg, other_args)
+
+    def _get_prepare_pydantic_annotations_for_known_type(self, obj: Any) -> Callable[..., Any] | None:
+        from decimal import Decimal
+
+        from . import _std_types_schema as std_types
+
+        if obj is Decimal:
+            return std_types.decimal_prepare_pydantic_annotations
+
+        return None
+
+    def prepare_annotations(self, source_type: Any, annotations: Iterable[Any]) -> tuple[Any, list[Any]]:
```

Comment:
I think this can be private, also can you add a docstring.

Reply:
It's already a private class. The only reason we have some public and some private things is that we used to at least partially expose this as part of a public interface. Now it's not, so we should just make everything public.

Reply:
I agree with you David. I added a note to make a PR _after_ this one that does renaming.